Description: don't compile third_party source
 .
Author: Laszlo Boszormenyi (GCS) <gcs@debian.org>
Forwarded: not-needed
Last-Update: 2022-09-28

---

--- protobuf-3.9.1.orig/Makefile.am
+++ protobuf-3.9.1/Makefile.am
@@ -12,7 +12,7 @@ SUBDIRS = . src
 #
 # Note that distribution artifacts will be produced by Bazel in the future.
 # See pkg/BUILD.bazel for overall definitions.
-DIST_SUBDIRS = src conformance benchmarks third_party/googletest
+DIST_SUBDIRS = src conformance benchmarks
 
 # Build gmock before we build protobuf tests.  We don't add gmock to SUBDIRS
 # because then "make check" would also build and run all of gmock's own tests,
--- protobuf-3.9.1.orig/configure.ac
+++ protobuf-3.9.1/configure.ac
@@ -241,7 +241,6 @@ AM_CONDITIONAL([OBJC_CONFORMANCE_TEST],
 #   too.
 export CFLAGS
 export CXXFLAGS
-AC_CONFIG_SUBDIRS([third_party/googletest])
 
 AC_CONFIG_FILES([Makefile src/Makefile benchmarks/Makefile conformance/Makefile protobuf.pc protobuf-lite.pc])
 AC_OUTPUT
